### PR TITLE
handles multiple lookup field values

### PR DESCRIPTION
**Pull request by HiteshRepo**
handles multiple lookup field values.

**Original `Lookup` list with single value**:
Lookup-List-Single

**Restored `Lookup` list with single value**:
Restored-Lookup-List

**Original `Lookup` list with multiple values**:
Lookup-List-Multi

**Restored `Lookup` list with multiple values**:
Restored-Lookup-List-Multi

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)
#5108 
#5084 

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/api/lists.go [hunk 1]
```diff
@@ -595,7 +620,7 @@ func specifyODataType(filteredData map[string]any, colDetails *columnDetails, co
 	}
 
 	switch {
-	case colDetails.isPersonColumn:
+	case colDetails.isPersonColumn || colDetails.isLookupColumn:
```
Comment (pandeyabs) [inline src/pkg/services/m365/api/lists.go:623]: `,` should work too here. 

```suggestion
	case colDetails.isPersonColumn, colDetails.isLookupColumn:
```

**Review by pandeyabs — APPROVED** [review]

File: src/pkg/services/m365/api/lists.go [hunk 2]
```diff
@@ -420,6 +421,18 @@ func setColumnType(
 	case orig.GetNumber() != nil:
 		newColumn.SetNumber(orig.GetNumber())
 	case orig.GetLookup() != nil:
+		colDetails.isLookupColumn = true
+		isMultipleEnabled := ptr.Val(orig.GetLookup().GetAllowMultipleValues())
+		colDetails.isMultipleEnabled = isMultipleEnabled
+
+		if isMultipleEnabled {
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists.go:428]: same as the other PR, this can get reduced.

File: src/pkg/services/m365/api/lists.go [hunk 3]
```diff
@@ -547,9 +561,20 @@ func populateMultipleValues(val any, filteredData map[string]any, colDetails *co
 
 	lookupIDs := make([]float64, 0)
 
+	checkFields := func(colDetails *columnDetails) []string {
+		if colDetails.isLookupColumn {
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists.go:565]: a switch statement might be better here.

Also, why an ad-hoc func?  There's nothing here dependent upon runtime state.  Seems like you can move it to an official func and give it some quick unit tests.

File: src/pkg/services/m365/api/lists.go [hunk 4]
```diff
@@ -547,9 +561,20 @@ func populateMultipleValues(val any, filteredData map[string]any, colDetails *co
...
+			return []string{LookupIDKey, LookupValueKey, PersonEmailKey}
+		}
+
+		return []string{}
+	}
+
 	for _, nestedFields := range multiNestedFields {
-		if md, ok := nestedFields.(map[string]any); ok && keys.HasKeys(md,
-			[]string{LookupIDKey, LookupValueKey, PersonEmailKey}...) {
+		if md, ok := nestedFields.(map[string]any); ok && keys.HasKeys(md, checkFields(colDetails)...) {
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists.go:577]: same cleanup recommendations here as the other PR

**Pull request merged**